packages/block-editor/src/components/block-popover/inbetween.js

```diff
@@ -91,66 +91,53 @@ function BlockPopoverInbetween( {
 		};
 	}, [ previousElement, nextElement, isVertical ] );
 
-	const getAnchorRect = useCallback( () => {
+	const popoverAnchor = useMemo( () => {
 		if ( ( ! previousElement && ! nextElement ) || ! isVisible ) {
-			return {};
+			return undefined;
 		}
 
 		const { ownerDocument } = previousElement || nextElement;
 
-		const previousRect = previousElement
-			? previousElement.getBoundingClientRect()
-			: null;
-		const nextRect = nextElement
-			? nextElement.getBoundingClientRect()
-			: null;
-
-		if ( isVertical ) {
-			if ( isRTL() ) {
-				return {
-					top: previousRect ? previousRect.bottom : nextRect.top,
-					left: previousRect ? previousRect.right : nextRect.right,
-					right: previousRect ? previousRect.left : nextRect.left,
-					bottom: nextRect ? nextRect.top : previousRect.bottom,
-					height: 0,
-					width: 0,
-					ownerDocument,
-				};
-			}
-
-			return {
-				top: previousRect ? previousRect.bottom : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.left,
-				right: previousRect ? previousRect.right : nextRect.right,
-				bottom: nextRect ? nextRect.top : previousRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
-		if ( isRTL() ) {
-			return {
-				top: previousRect ? previousRect.top : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.right,
-				right: nextRect ? nextRect.right : previousRect.left,
-				bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
 		return {
-			top: previousRect ? previousRect.top : nextRect.top,
-			left: previousRect ? previousRect.right : nextRect.left,
-			right: nextRect ? nextRect.left : previousRect.right,
-			bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-			height: 0,
-			width: 0,
 			ownerDocument,
+			getBoundingClientRect() {
+				const prevRect = previousElement
+					? previousElement.getBoundingClientRect()
+					: null;
+				const nextRect = nextElement
+					? nextElement.getBoundingClientRect()
+					: null;
+
+				let left = 0;
+				let top = 0;
+
+				if ( isVertical ) {
+					// vertical
+					top = prevRect ? prevRect.bottom : nextRect.top;
+
+					if ( isRTL() ) {
+						// vertical, rtl
+						left = prevRect ? prevRect.right : nextRect.right;
+					} else {
+						// vertical, ltr
+						left = prevRect ? prevRect.left : nextRect.left;
+					}
+				} else {
+					top = prevRect ? prevRect.top : nextRect.top;
+
+					if ( isRTL() ) {
+						// non vertical, rtl
+						left = prevRect ? prevRect.left : nextRect.right;
+					} else {
+						// non vertical, ltr
+						left = prevRect ? prevRect.right : nextRect.left;
+					}
+				}
+
+				return new window.DOMRect( left, top, 0, 0 );
+			},
 		};
-	}, [ previousElement, nextElement ] );
+	}, [ previousElement, nextElement, isVisible ] );
```

Comment:
Change no. 3: add `isVisible` to the list of dependencies

Reply:
🙈 

https://github.com/WordPress/gutenberg/pull/43693/commits/4d6926aada3a9f777dfb1a718b0ef13739c53c42